src/viewer.js

```diff
@@ -41,6 +42,25 @@ class Viewer {
 
     /** @private {?Element} */
     this.iframe_ = null;
+
+    /** @private {!History} */
+    this.history_ = new History(this.unAttach.bind(this));
+  }
+
+  /**
+   * @param {!Function} hideViewer method that hides the viewer.
+   */
+  setViewerHider(hideViewer) {
```

Comment:
I feel like there are going to be lots of things to pass to the constructor going forward and it would be cleaner to have setter methods